data/pools/transactionPool.go

```diff
@@ -323,7 +373,7 @@ func (pool *TransactionPool) checkSufficientFee(txgroup []transactions.SignedTxn
 	// get the current fee per byte
 	feePerByte := pool.computeFeePerByte()
 
-	for _, t := range txgroup {
+	for _, t := range txgroup.Transactions {
```

Comment:
Not related to txnsync, but my feeling is that this will not work well with the new accounting option to share the fees among transactions in the same transaction group.

Reply:
I agree - we might want to revisit this outside the scope of the transaction sync.

Reply:
I created a ticket. https://github.com/algorand/go-algorand/issues/2751